Src/DependencyCollector/Microsoft.AI.DependencyCollector.Tests/DependencyCollectorDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,251 @@
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+
+    [TestClass]
+    public class DependencyCollectorDiagnosticListenerTests
+    {
+        private string instrumentationKey;
+        private StubTelemetryChannel telemetryChannel;
+        private DependencyCollectorDiagnosticListener listener;
+
+        private List<ITelemetry> sentTelemetry = new List<ITelemetry>();
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            instrumentationKey = Guid.NewGuid().ToString();
+
+            telemetryChannel = new StubTelemetryChannel()
+            {
+                EndpointAddress = "https://endpointaddress",
+                OnSend = sentTelemetry.Add
+            };
+
+            listener = new DependencyCollectorDiagnosticListener(new TelemetryConfiguration()
+            {
+                TelemetryChannel = telemetryChannel,
+                InstrumentationKey = instrumentationKey,
+            });
+        }
+
+        [TestMethod]
+        public void OnRequestWithRequestEventWithNoRequestUri()
+        {
+            listener.OnRequest(new HttpRequestMessage(), Guid.NewGuid());
+            Assert.AreEqual(0, listener.PendingDependencyTelemetry.Count());
+            Assert.AreEqual(0, sentTelemetry.Count);
+        }
+
+        [TestMethod]
+        public void OnRequestWithRequestEvent()
+        {
+            Guid loggingRequestId = Guid.NewGuid();
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "https://www.example.com");
+            listener.OnRequest(request, loggingRequestId);
+
+            Assert.AreEqual(1, listener.PendingDependencyTelemetry.Count());
+            DependencyTelemetry telemetry = listener.PendingDependencyTelemetry.Single();
+            Assert.AreEqual("POST /", telemetry.Name);
+            Assert.AreEqual("www.example.com", telemetry.Target);
+            Assert.AreEqual("Http", telemetry.Type);
+            Assert.AreEqual("https://www.example.com", telemetry.Data);
+            Assert.AreEqual("", telemetry.ResultCode);
+            Assert.AreEqual(true, telemetry.Success);
+
+            Assert.AreEqual(InstrumentationKeyHashLookupHelper.GetInstrumentationKeyHash(instrumentationKey), GetRequestHeaderValues(request, DependencyCollectorDiagnosticListener.SourceInstrumentationKeyHeader).SingleOrDefault());
+            Assert.IsFalse(GetRequestHeaderValues(request, DependencyCollectorDiagnosticListener.StandardRootIdHeader).Any());
+            Assert.IsTrue(GetRequestHeaderValues(request, DependencyCollectorDiagnosticListener.StandardParentIdHeader).Any());
+
+            Assert.AreEqual(0, sentTelemetry.Count);
+        }
+
+        private static IEnumerable<string> GetRequestHeaderValues(HttpRequestMessage request, string headerName)
```

Comment:
I've built using buildRelease.cmd, which sets StyleCopEnabled to True, so I believe I've gotten this to pass StyleCop.